board/safety/safety_subaru.h

```diff
@@ -186,9 +222,12 @@ static int subaru_fwd_hook(int bus_num, int addr) {
 
 static const addr_checks* subaru_init(uint16_t param) {
   subaru_gen2 = GET_FLAG(param, SUBARU_PARAM_GEN2);
+  subaru_gen3 = GET_FLAG(param, SUBARU_PARAM_GEN3);
 
   if (subaru_gen2) {
     subaru_rx_checks = (addr_checks){subaru_gen2_addr_checks, SUBARU_GEN2_ADDR_CHECK_LEN};
+  } else if (subaru_gen3) {
+    subaru_rx_checks = (addr_checks){subaru_gen3_addr_checks, SUBARU_GEN3_ADDR_CHECK_LEN};
```

Comment:
Do we know if Subaru is calling this car having a gen 2 EyeSight system? Where did we get the current gen 2 name? Would like to make the name appropriate. Right now there's just a few differences. Perhaps these are still gen 2 cars, just with alternate LKAS and cruise activated signals, like we handle Hyundai's differences.

Reply:
I think they are similar enough to where it doesn't make sense to make new files, but flags might make sense

SUBARU_ALT_LKAS
(Only this car has this arm, but I suspect it will also be on the other cars with the new harnesses)

and

SUBARU_USE_ES_STATUS
(Cruise_Activated was removed from CruiseControl, need to use ES_STATUS instead, unless we can find another bit from the PCM that matches our old CruiseControl bit. Other cars have this problem too, such as the hybrids.)

Reply:
gen2 refers more to new canbus architecture with some critical
messages on 2 redundant buses. Eyesight can messages have not changed much in the process and can use messsage/feature flags

Reply:
https://github.com/martinl/openpilot/wiki/CAN-Network-diagrams